reliability/node_tuning_operator.py

```diff
@@ -0,0 +1,344 @@
+
+repeats = 0
+while True:
+    print("Attempt #{}/12 starts in 10 seconds.".format(repeats + 1))
+    for node in nodes:
+        kernel_pid_max_on_node = int(execute_command_on_node(external_dns, node, "sysctl kernel.pid_max").rstrip().split("=")[1])
+        if kernel_pid_max_before != kernel_pid_max_on_node:
+            fail("On node {} kernel.pid_max is {} instead {}".format(node, kernel_pid_max_on_node, kernel_pid_max_after))
+        else:
+            print("Match on node: {} with value: {}".format(node, kernel_pid_max_on_node))
+    repeats += 1
+    if repeats == 12:
+        break
+    time.sleep(10)
+
+print("Checking logs of tuned pods on each node after test")
+tuning_applied_after = []
+for pod in tuned_pods:
+    tuning_applied_after.append(int(execute_command("oc logs {} | grep -c applied".format(pod))))
+
+print("\nResults:")
+print("Pod\t\tBefore\tAfter")
+for i in range(len(tuned_pods)):
+    print("{}\t{}\t{}".format(tuned_pods[i], tuning_applied_before[i], tuning_applied_after[i]))
+
+for i in range(len(tuning_applied_before)):
+    if tuning_applied_after[i] == tuning_applied_before[i]:
+        fail("At least one of pods should be tuned")
+passed(None)
+
+
+# Verification that changing priority affect at least one node
+print_step("Verify that after changing priority pod will be tuned")
+print("Checking logs of tuned pods on each node before test")
+tuning_applied_before = []
+for pod in tuned_pods:
+    tuning_applied_before.append(int(execute_command("oc logs {} | grep -c applied".format(pod))))
+
+# Saving default configuration to file
+print_step("Saving default configuration file")
+execute_command("oc get tuned default -o yaml > default_values.yaml")
+execute_command("sed -e \"s/ priority: 20/ priority: 15/\" default_values.yaml > new_priority.yaml")
+execute_command("oc apply -f new_priority.yaml")
+print("Waiting 15 seconds, to be sure changes has applied")
+time.sleep(15)
+
+tuning_applied_priority_after = []
+for pod in tuned_pods:
+    tuning_applied_priority_after.append(int(execute_command("oc logs {} | grep -c \"\'openshift-node-es\' applied\"".format(pod))))
+print("\nResults:")
+print("Pod\t\tBefore\tAfter")
+for i in range(len(tuned_pods)):
+    print("{}\t{}\t{}".format(tuned_pods[i], tuning_applied_before[i], tuning_applied_priority_after[i]))
+test_pass = False
+for i in range(len(tuning_applied_before)):
+    if tuning_applied_priority_after[i] > tuning_applied_before[i]:
+        test_pass = True
+        passed(None)
+        break
+if not test_pass:
+    fail("At least one of pods should be tuned")
+
+# Cleaning after execution test
+execute_command("oc delete project my-logging-project")
+print_step("Saving default configuration file")
+execute_command("oc get tuned default -o yaml > default_values.yaml")
+execute_command("sed -e \"s/ priority: 15/ priority: 20/\" default_values.yaml > new_priority.yaml")
```

Comment:
I would save the original file somewhere and use it to recover, instead of maintaining to change back what has been changed.

Reply:
Yes, I wanted to do the same... But every time when you load a new file, version number of default settings is changing, so original file need to be updated with correct version number.
So the choice was between modifying version number or changing values.

Reply:
Did you mean something like this?

```
for: "/home/fedora/abc.td.yaml": Operation cannot be fulfilled on tuneds.tuned.openshift.io "default": the object has been modified; please apply your changes to the latest version and try again
```

Probably this is a bug.
I would suggest:
Use the current value `metadata.resourceVersion` to replace the one in the original yaml, then apply the original yaml.

Reply:
`oc delete tuned default`
`oc create -f ./default_values.yaml'
to restore correct version.